syntax = "proto3";

package payment;

service PaymentService {
  rpc Authorize (AuthorizeRequest) returns (AuthorizeResponse);
  rpc Capture (CaptureRequest) returns (CaptureResponse);
  rpc Refund (RefundRequest) returns (RefundResponse);
}

message AuthorizeRequest {
  string idempotency_key = 1;
  string tenant_id = 2;
  int64 amount = 3;
  string currency = 4;
}

message AuthorizeResponse {
  string payment_id = 1;
  string provider = 2;
  string provider_reference = 3;
  string state = 4;
}

message CaptureRequest {
  string payment_id = 1;
  int64 amount = 2;
}

message CaptureResponse {
  string payment_id = 1;
  string provider = 2;
  string provider_reference = 3;
  string state = 4;
}

message RefundRequest {
  string payment_id = 1;
  int64 amount = 2;
}

message RefundResponse {
  string payment_id = 1;
  string provider = 2;
  string provider_reference = 3;
  string state = 4;
}
